src/binder/bindertracing.cpp

```diff
@@ -231,3 +262,123 @@ namespace BinderTracing
         return m_ignoreBind;
     }
 }
+
+namespace BinderTracing
+{
+    ResolutionAttemptedOperation::ResolutionAttemptedOperation(BINDER_SPACE::AssemblyName *assemblyName)
+        : m_pAssemblyName{assemblyName}
+    {
+    }
+
+    ResolutionAttemptedOperation::~ResolutionAttemptedOperation()
+    {
+        TraceCurrentStage();
+    }
+
+    void ResolutionAttemptedOperation::SetResult(HRESULT hr, BINDER_SPACE::Assembly *assembly)
+    {
+        if (SUCCEEDED(hr))
+        {
+            m_result = Result::Success;
+        }
+        else if (hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
+        {
+            m_result = Result::AssemblyNotFound;
+        }
+        else if (hr == FUSION_E_APP_DOMAIN_LOCKED)
+        {
+            m_result = Result::IncompatibleVersion;
+        }
+        else if (hr == FUSION_E_REF_DEF_MISMATCH)
+        {
+            m_result = Result::MismatchedAssemblyName;
+        }
+        else
+        {
+            m_result = Result::Unknown;
+            UNREACHABLE();
+        }
+
+        m_pFoundAssembly = assembly;
+    }
+
+    void ResolutionAttemptedOperation::Trace(Stage stage, INT_PTR managedALC, AssemblySpec *assemblySpec)
+    {
+        m_pAssemblySpec = assemblySpec;
+        m_pManagedALC = managedALC;
+        m_stage = stage;
+        TraceCurrentStage();
+    }
+
+    void ResolutionAttemptedOperation::TraceCurrentStage()
+    {
+        if (m_stage == Stage::NotYetStarted)
+        {
+            return;
+        }
+
+        ResolutionAttempt attempt;
+
+        attempt.AssemblyName = m_pAssemblyName->GetSimpleName();
+        attempt.Stage = static_cast<uint16_t>(m_stage);
+        attempt.Result = static_cast<uint16_t>(m_result);
+
+        if (m_pManagedALC != 0)
+        {
+            GetAssemblyLoadContextNameFromManagedALC(m_pManagedALC, attempt.AssemblyLoadContext);
+        }
+        else if (m_pAssemblySpec != nullptr)
+        {
+            GetAssemblyLoadContextNameFromSpec(m_pAssemblySpec, attempt.AssemblyLoadContext);
+        }
+        else
+        {
+            attempt.AssemblyLoadContext.Set(W("Default"));
```

Comment:
The TPA binder gets updated with a pointer to the managed default ALC (when it is created), so `m_pManagedALC` would not be 0 at that point - we'd still want that to be logged as 'Default'.